File: learning/链式编程/链式编程/ViewController.m
//
//  ViewController.m
//  链式编程
//
//  Created by WZH on 16/8/19.
//  Copyright © 2016年 王振海. All rights reserved.
//

#import "ViewController.h"
#import "ZHChainedLabel.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
   
    ZHChainedLabel *label = [[ZHChainedLabel alloc] init];
    label.ZHFrame(CGRectMake(100, 100, 100, 100)).ZHFont(15).ZHText(@"haha").ZHColor([UIColor redColor]).addToSuperview(self.view);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Demo/Swift/26.Swift3.0 - 懒加载/26.Swift3.0 - 懒加载/ViewController.swift
//
//  ViewController.swift
//  26.Swift3.0 - 懒加载
//
//  Created by WZH on 16/10/19.
//  Copyright © 2016年 limpid. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    //只是定义，没有分配空间和初始化
    
    //什么时候用as转换 -> 父类转换为子类
    var label: DemoLabel?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    private func setIp() {
        
        //1.创建空间
        label = DemoLabel()
        
        //2.添加到视图
        //! 解包，为了参与计算， addSubView，用subviews 数组记录空间，数组中不允许加入nil
        //? ---- 可选解包，调用方法，如果为nil，不调用方法，但是不能参与计算
        view.addSubview(label!)
        
        label?.text = "hello"
        label?.sizeToFit()
        label?.center = view.center
    }


}

